### DISCUSSION TITLE
How to type select property

### DISCUSSION THREAD

**Discussion by Mathie01** [prisma/prisma#12903] [Q&A]
Hello !

I'm using Prisma for the first time and I can't understand the documentation of `findUnique()` : 

![image](https://user-images.githubusercontent.com/67562521/164314367-aa15acfc-5f04-41ee-87d5-18ccfced574c.png)

How to type `select ` in my code ? :')
I can't set `Xor<UserSelect | null>`

Edit : I was able to use the `Prisma.UserSelect` type !
But I don't know what type to put in the return of my findUnique() function. I can't type it as 'User' if I add a Select

```
sync user(
        userWhereUniqueInput: Prisma.UserWhereUniqueInput,
        select?: Prisma.UserSelect,
    ): Promise<**?**> {
        return this.client.prisma.user.findUnique({
            where: userWhereUniqueInput,
            select: select
        })
}
```

Can you help me ? 
Thanks !!

**Comment by nurul3101** [comment]
Hey @Mathie01 👋 

You are looking for using `PromiseReturnType` that is exposed by Prisma namespace for exactly these kinds of use cases.
Here's a detailed example of how you can get the return types when doing `select` or `include` operations: [Getting Access to return type of function](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-getting-access-to-the-return-type-of-a-function)

Can you please have a look at it and see if this solves it for you?

**Comment by Mathie01** [comment]
Hello !

How can I use `type` in class ?

![image](https://user-images.githubusercontent.com/67562521/166147216-4306a69a-f31d-4978-af21-56883fe5fd47.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
